src/Tokenizer/TokensAnalyzer.php

```diff
@@ -526,10 +526,11 @@ public function isWhilePartOfDoWhile($index)
     /**
      * @param Tokens $tokens
      * @param int    $start
+     * @param bool   $searchingOnFileLevel if searching on file level
```

Comment:
not sure here. like if we have class A, inside of which we have anon class B, inside of which we have anon class C...
when we call function to investigate class B and it find out about class C so it is just about to call investigation for class C, this 3rd investigation is not willing to investigate file-level class A indeed, but also it is not supposed to investigate class B as well

Reply:
it is a naming thing, if C is in B which is in A, C won't investigate B (anymore), it is more like a flag since only on file level you might have multiple classes you want to check on the same level, inside a class you never want to look outside it

Reply:
:+1:
i double check the code, it's like you say.
so now, we just need to find proper name for it.

Reply:
What about `$stopAfterCurrentClass` (would require to switch the boolean values)?

Reply:
I reworked the methods to only check each token one time, WDYT @keradus @julienfalque ?

Reply:
:+1: with minors.

Reply:
wow, I just saw that. from first glance looks OKish, yet it's important part of code, I need to checkout branch and play with the change

Reply:
thanks man :+1: ,
if it is to much trouble I roll back the last commit

Reply:
nah, code looks really better now. i just need to analyse it...